### ISSUE TITLE
use 100% cpu with flatpak installation on alpine

### ISSUE THREAD

**Issue by somethingGoneWrong** [flxzt/rnote#261]
**Describe the bug**
the program works well, I use it on a samsung tablet[sm-t550] with postmarketos installed [gnome/phosh], but it uses 100% CPU of all 4 cores.

**To Reproduce**
1. run flatpak run -v com.github.flxzt.rnote from terminal
2. use it normal
3. check CPU usage

**Expected behavior**
a normal use of CPU, I use Firefox latest version, and it not consumes all of cores

** Console Output **
```
samsung_gt510:~$ flatpak run -v com.github.flxzt.rnote  
F: No installations directory in /etc/flatpak/installations.d. Skipping
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/user/.local/share/flatpak
F: Opening user flatpak installation at path /home/user/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/user/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: /var/lib/flatpak/runtime/org.gnome.Platform/aarch64/42/f0bb984a14e33b7d16b0c52c4c7f2d179c02e07c8faf18a9bca61babf0939694/files/lib32 does not exist
F: /var/lib/flatpak/runtime/org.gnome.Platform/aarch64/42/f0bb984a14e33b7d16b0c52c4c7f2d179c02e07c8faf18a9bca61babf0939694/files/lib64 does not exist
F: Cleaning up unused container id 651613085
F: Cleaning up per-app-ID state for com.github.flxzt.rnote
F: Allocated instance id 2042065093
F: Add defaults in dir /com/github/flxzt/rnote/
F: Add locks in dir /com/github/flxzt/rnote/
F: Disallowing network access
F: Allowing dri access
F: Allowing wayland access
F: Allowing x11 access
F: Allowing pulseaudio access
F: Pulseaudio user configuration file '/home/user/.config/pulse/client.conf': Error opening file /home/user/.config/pulse/client.conf: No such file or directory
F: Could not find pulseaudio socket
F: Failed to run in transient scope: No systemd user session available, cgroups not available
F: Running 'bwrap --args 33 rnote xdg-dbus-proxy --args=38'
F: Running 'bwrap --args 33 rnote'
```

**Screenshots**
I attached a screenshot of htop tool and some useful logs

**Desktop (please complete the following information):**
 - OS: postmarketOS [alpine linux]
 - Version: 22.06.2
 - Installation source: flatpak
 - Desktop Environment: Gnome 44.4

![rnote_issue](https://user-images.githubusercontent.com/40141099/192145096-3ca2d822-a808-48bf-8d54-3be61081e399.png)
[log_rnote.txt](https://github.com/flxzt/rnote/files/9640812/log_rnote.txt)

**Comment by flxzt** [comment]
Nice to hear from a user on a tablet! If you are noticing any paper cuts or improvement ideas specific to usage on a tablet, please don't hesitate to open issues for it.

Regarding the performance. Rnote is rerendering the visible area asynchronously and in parallel after a zoom gesture has finished (You'll notice that the textures of strokes and images become sharper after a short while after a zoom). It should stabilize however and usage should go down eventually. 

It obviously also depends on the number of strokes visible on the screen. For example, many vector images can increase the render time, so can a lot of large strokes (although there is a mechanism implemented to split such a large stroke into many smaller textures).

Does it stay at 100% all the time, or are there only usage spikes after such an event? If you are experiencing the former, it sounds like a bug that must be fixed, if the latter is happening, it would fall under the category of performance optimization.

**Comment by Xelef2000** [comment]
I have a similar experience with my Surface Go 3 (Intel Pentium GOLD 6500Y) on Ubuntu 22.04 and power mode "Balanced".
Even in small documents (3-5 pages), rnote becomes sluggish and 3 cores hover between 70% and 100%. It also rapidly drains my battery. Rnote becomes unusable on the power mode "power saver".
Both issues are so severe that it prevents me from using rnote on a daily basis.
I don't have the same issues on my much more powerful laptop.

I don't think this is a bug, but I really would like some performance optimizations for low-power CPUs or even a low-power mode.

**Comment by somethingGoneWrong** [comment]
@flxzt 
> Does it stay at 100% all the time, or are there only usage spikes after such an event? If you are experiencing the former, it sounds like a bug that must be fixed, if the latter is happening, it would fall under the category of performance optimization.

yeah, it was always at 100% cpu usage, but I try to replicate the issue without success, now seems to works well, I need to investigate more about what generated that situation.
Do you have any useful commands to run to get more useful information about the process if this happens again?

> Nice to hear from a user on a tablet! If you are noticing any paper cuts or improvement ideas specific to usage on a tablet, please don't hesitate to open issues for it.

I found the UI really convenient and well-made, it fit the size of the 9.7" tablet very well; sometimes popmenu widgets appear and disappear by themselves in milliseconds, I think it is a GTK/libadwaita problem as it also happens when I use Firefox and to a more serious way. 
For now only one thing I found it inconvenient, and I don't know if it's a bug or something else; when I'm drawing, and I want to zoom with the two finger gestures, in or out it doesn't make a difference, the program performs correctly, but it draws on the points of the two touches one point and on the other a line that I think traces the movement performed for the zoom.
Annoying, but not a serious problem.

I have some ideas about expanding the already great features, I was thinking of something similar fiinote you find on android, and since I would like to get my hands dirty with rust it would be a great way to learn something new. I will email you to see if these possible new features would be of interest to you or not.

@Xelef2000 
> It also rapidly drains my battery. Rnote becomes unusable on the power mode "power saver".

I don't know if there is a power saving mode on posh i have to investigate.

> I don't think this is a bug, but I really would like some performance optimizations for low-power CPUs or even a low-power mode.

I agree

**Comment by flxzt** [comment]
Another user reported a very similar issue on Matrix.
After profiling the app we found that the cause of high CPU load is actually in the audio feature of the app. I believe this could be the cause for all reports of high CPU usage here, so it would be a bug. Similar things are reported in the dependency crates that are used in rnote. Notably issues: https://github.com/RustAudio/rodio/issues/431 and https://github.com/RustAudio/cpal/issues/730 .
It resolved for him after switching from pulseaudio to pipewire as audio backend, but that may not be a possibility for everyone.
A possible upstream fix could be https://github.com/RustAudio/cpal/pull/731 , but it would still need testing if this actually resolves it.
Until then I changed the behaviour of the `pen sounds` setting so that the audio streams are actually only initialized once it is enabled (in 945435759d9b7bdb274fcd24584b84f873af2d21 ). So for anyone experiencing the same problem, it should be enough to simply keep this setting disabled for now.
If it then still occurs, we at least ruled out this issue as the cause of it.
